### PR TITLE
[#658] Added suite-level accessibility aggregate report.

### LINKED ISSUE #658: Add a cross-page accessibility report aggregated across the whole suite
## Summary

`AccessibilityTrait` writes one HTML + JUnit report per scenario. There is no single, suite-level report that aggregates accessibility results across every page assessed during a run. To understand the overall accessibility state of a site you currently have to open and reconcile N per-scenario files.

Request: produce one combined cross-page report at the end of the run, in addition to the existing per-scenario reports.

## Current behaviour

`src/AccessibilityTrait.php` (3.10.0):

- `accessibilityResults` is a per-instance array, reset for every scenario in `#[BeforeScenario] accessibilitySetupScenario()`.
- `#[AfterScenario] accessibilityFinalizeScenario()` writes `<feature>__<scenario>.html` and `junit-<feature>__<scenario>.xml` for that scenario only.
- There is no static accumulator across scenarios and no `@AfterSuite` / `@AfterFeature` hook, so nothing combines the per-scenario results.

Result: a run that assesses, say, the homepage, `/contact`, and `/admin` across ten scenarios yields ten separate reports, the same page (e.g. the homepage) is reported repeatedly, and there is no single rollup.

## Desired behaviour

A single aggregate report written once at the end of the run (e.g. `index.html` plus an aggregate `junit.xml`) that:

- Lists every URL assessed, de-duplicated across scenarios.
- Groups violations by page and by rule, with counts per impact level.
- Surfaces a run-level summary (total pages, total unique violations, breakdown by impact).
- Links to or embeds the per-scenario detail.

## Possible implementation

- Accumulate normalized results into a `static` property as each scenario finalizes (keyed by URL + rule, retaining feature / scenario provenance).
- Add an `#[AfterSuite]` static hook that renders the combined report from the accumulator into the report directory.
- Provide override points mirroring the existing per-scenario render methods (e.g. `accessibilityRenderAggregateHtml()` / `...Page()` / `...Sections()`), so the aggregate page can be rebranded without rewriting the aggregation logic.

Behat supports static `@AfterSuite` hooks, so this can live entirely within the trait - no separate Behat extension or external collector required.

## Why in the library

Consumers otherwise have to build a custom Behat extension or a post-processing collector to merge the per-scenario JUnit / HTML files. Since the trait already owns the normalized result shape and the renderers, a suite-level rollup is a natural extension and avoids every consumer reinventing it.

## Related

- #657 (report directory location) - the aggregate report should honour the same configurable report directory.

## Environment

- drevops/behat-steps 3.10.0

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
Closes #658

## Summary

Adds a single suite-level accessibility report to `AccessibilityTrait`, written once after the whole Behat suite via a static `#[AfterSuite]` hook. Each scenario still writes its own per-scenario HTML and JUnit reports; the new aggregate collects and de-duplicates findings across every scenario into one cross-page overview. It is written to `accessibility_report_<YYYYMMDD_HHMMSS>.html` in the configured report directory - one timestamped file per run, so a run never overwrites a previous one. The work reuses the report directory and URL-formatting helpers from #657 and #659, so consumer overrides of those are honoured.

## Changes

**Accumulation**
- `$accessibilityAggregate` (process-global) collects each scenario's results; `$accessibilityAggregateReportDir` captures the resolved report directory from the instance phase for the static renderer.
- `accessibilityAggregateCapture()` records each scenario's formatted results (URLs passed through `accessibilityFormatUrl()`) in `accessibilityFinalizeScenario()`, before the gate throw, so a scenario that fails the gate is still represented.
- `accessibilityAggregateReset()` (`#[BeforeSuite]`) clears the statics so a second suite in the same process starts clean.

**Calculation - one method per concern**
- `accessibilityAggregatePages()` de-duplicates assessed pages by URL, keyed by `feature > scenario` so same-titled scenarios keep their provenance.
- `accessibilityAggregateRollup()` rolls violations up by rule, tallies by impact, and sorts by severity then affected-element count.
- `accessibilityAggregateFindings()` flattens findings and stringifies element targets.
- `accessibilityAggregateData()` is the single calculation entry point: it calls the above and assembles one render-ready data array (`generated`, `page_count`, `scenario_count`, `total_violations`, `totals`, `pages`, `rules`, `scenarios`).

**Rendering - one method**
- `accessibilityRenderAggregate(array $data)` is the single rendering entry point. It receives the prepared data array and produces the entire HTML document: summary cards, a de-duplicated "Pages assessed" table with per-rule chips, a severity-sorted "Violations by rule" rollup, full per-scenario detail, and the CSS. Override this one method to restyle the whole report without touching any aggregation logic.

**Reporting and helpers**
- `accessibilityAggregateRender()` (`#[AfterSuite]`) and `accessibilityWriteAggregateReport()` resolve the timestamp once - it drives both the filename and the in-page "generated" line - and write the report.
- `accessibilityBlankUrls()` (shared, overridable) adds `data:,` to the blank-URL skip list, used by both per-step auto-assessment and the aggregate.
- `accessibilityStringifyTarget()` is promoted to `protected static` so the static renderer can reuse it.
- Trait methods are grouped by concern: all lifecycle hooks together, the format helpers together, and the aggregate calculation + render block together.

**Tests**
- PHPUnit unit tests cover each calculation method, the single render method (driven directly from a hand-built data array), the writer (timestamped filename), the capture/reset hooks, and edge cases (clean run, empty `helpUrl`, blank-URL skipping).
- A Behat `@trait:AccessibilityTrait` integration scenario asserts the timestamped `accessibility_report_*.html` is written after a real sub-process suite run, via a new generic `a file matching "<glob>" should exist` step.

## Before / After

The per-scenario "single page" report machinery is unchanged. This PR adds the suite aggregate and wires it to reuse the same small helpers (`accessibilityFormatUrl()`, `accessibilityStringifyTarget()`, `accessibilityBlankUrls()`) rather than re-implementing them.

**Before** - only the per-scenario "single page" report exists:

```text
#[AfterScenario] accessibilityFinalizeScenario()
   |
   |-- accessibilityRenderHtml()                  ->  <feature>__<scenario>.html
   |      |-- accessibilityRenderHtmlPage()            page wrapper + CSS
   |      `-- accessibilityRenderHtmlSections()        one section per visited URL
   |             |-- accessibilityFormatUrl()              URL -> path
   |             `-- accessibilityRenderIssueList()        violations + incomplete
   |                    `-- accessibilityStringifyTarget()     flatten node target
   |
   `-- accessibilityRenderJunit()                 ->  junit-<feature>__<scenario>.xml
          |-- accessibilityFormatUrl()
          `-- accessibilityStringifyTarget()

formatUrl() and stringifyTarget() are instance helpers used only by this
per-scenario path. There is no cross-page / suite-level report.
```

**After** - the per-scenario report is unchanged; a suite aggregate is added, and both reports reuse the same helpers:

```text
shared helpers (reused by both reports)
   accessibilityFormatUrl()        URL -> path
   accessibilityStringifyTarget()  flatten node target   (promoted to static so the
   accessibilityBlankUrls()        blank-tab skip list      static aggregate can call it)

#[AfterScenario] accessibilityFinalizeScenario()
   |
   |-- accessibilityRenderHtml() + accessibilityRenderJunit()  ->  per-scenario files
   |      `-- reuse formatUrl(), stringifyTarget()                  (unchanged)
   |
   `-- accessibilityAggregateCapture()                         [NEW]
          `-- reuse formatUrl(); append results to a static accumulator
                                                                      |
                                                                      v
#[AfterSuite] accessibilityAggregateRender()                    [NEW]
   |
   `-- accessibilityWriteAggregateReport()  ->  accessibility_report_<YYYYMMDD_HHMMSS>.html
          |
          |-- accessibilityAggregateData()              CALCULATION (one entry point)
          |      |-- accessibilityAggregatePages()      reuse blankUrls()
          |      |-- accessibilityAggregateRollup()     reuse stringifyTarget()
          |      `-- accessibilityAggregateFindings()   reuse stringifyTarget()
          |      returns $data = [generated, totals, pages, rules, scenarios, ...]
          |
          `-- accessibilityRenderAggregate($data)       RENDERING (one method -> full HTML)
```

## Screenshots

Accessibility aggregate report

### CODE REVIEW

File: src/AccessibilityTrait.php [hunk 1]
```diff
@@ -887,4 +918,524 @@ protected function accessibilityRenderJunit(): string {
...
+          continue;
+        }
+
+        if (!isset($pages[$url])) {
+          $pages[$url] = [
+            'violations' => $result['result']['violations'] ?? [],
+            'incomplete' => count($result['result']['incomplete'] ?? []),
+            'passes' => count($result['result']['passes'] ?? []),
+            'scenarios' => [],
+          ];
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/AccessibilityTrait.php:1052]: Not adopting this one. `array_merge` would multi-count a page's findings whenever more than one scenario visits the same URL: the common case is that every visit of a given URL yields the same violations, so merging would repeat each violation (and inflate the per-impact totals) once per visiting scenario. The "Pages assessed" table is deliberately a de-duplicated, one-row-per-URL summary, and the complete per-visit findings - including any rule or threshold differences between scenarios - are preserved verbatim in the "Per-scenario detail" section, so nothing is hidden from the report. Keeping the first-seen result is the intended de-dup behaviour.

File: src/AccessibilityTrait.php [hunk 2]
```diff
@@ -887,4 +918,524 @@ protected function accessibilityRenderJunit(): string {
...
+        '<div class="rule"><h3><span class="impact %s">%s</span> <span class="rule-id">%s</span></h3>'
+        . '<p class="meta">%s &middot; affects %d page(s) &middot; %d element(s) &middot; <a href="%s" target="_blank" rel="noopener">docs</a></p>%s</div>',
+        htmlspecialchars($impact, ENT_QUOTES),
+        htmlspecialchars($impact, ENT_QUOTES),
+        htmlspecialchars((string) $rule_id, ENT_QUOTES),
+        htmlspecialchars((string) $rule['help'], ENT_QUOTES),
+        count($rule['pages'] ?? []),
+        count($rule['nodes'] ?? []),
+        htmlspecialchars((string) $rule['helpUrl'], ENT_QUOTES),
+        $nodes
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/AccessibilityTrait.php]: Fixed in 2ab9744. The rule rollup now builds the docs anchor only when `helpUrl` is non-empty, matching the existing guard in the per-scenario report and in the aggregate's per-scenario issue list, so an empty `helpUrl` no longer emits `href=""`. Output is unchanged for rules that have a docs URL. Added a unit test (`testAggregateRulesOmitDocsLinkWhenHelpUrlEmpty`) asserting no empty `href` is produced.

File: STEPS.md [hunk 3]
```diff
@@ -78,6 +78,12 @@
 >  be plugged in by overriding `accessibilityRunEngine()` (perform the
 >  assessment, return raw results) and `accessibilityNormalizeResults()`
 >  (remap raw output into the canonical shape the rest of the trait expects).
+>  <br/><br/>
+>  Reporting. Each scenario writes its own HTML and JUnit report. After the
+>  whole suite, a single cross-page `index.html` is written to the same
+>  directory, de-duplicating every assessed page and rolling violations up by
+>  rule. The aggregate accumulates in process-global state, so under parallel
+>  Behat each process writes its own `index.html`.
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply STEPS.md]: Not adopting this one. `STEPS.md` is generated by `docs.php` (regenerated via `ahoy update-docs`), which emits two spaces after `>` for every blockquote line in the document. Editing only these lines would make them inconsistent with the rest of the generated file and would fail `ahoy lint-docs` (`docs.php --fail-on-change`), which asserts that `STEPS.md` matches the generator output. MD027 is not part of this repo's lint pipeline.

File: tests/behat/features/accessibility.feature [hunk 4]
```diff
@@ -86,3 +86,17 @@ Feature: Check that AccessibilityTrait works
...
+  @trait:AccessibilityTrait
+  Scenario: A cross-page aggregate report is written after the suite
+    Given some behat configuration
+    And scenario steps tagged with "@javascript @accessibility-warning":
+      """
+      Given I visit "/sites/default/files/accessibility_violations.html"
+      Then I should see "Inaccessible Page"
+      When I visit "/sites/default/files/accessibility_clean.html"
+      Then I should see "Clean Accessibility Page"
+      """
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply tests/behat/features/accessibility.feature:99]: Not adopting this one. The triple-single-quote convention applies only to an inner PyString nested inside a step that itself takes a PyString argument. This scenario has no nested PyString - every inner step (`Given I visit`, `Then I should see`, `When I visit`) is single-line - so the outer block correctly uses `"""`, exactly like every other `@trait` scenario in this feature file. Switching to `'''` here would diverge from that established pattern without need.

File: src/AccessibilityTrait.php [hunk 5]
```diff
@@ -887,4 +953,455 @@ protected function accessibilityRenderJunit(): string {
...
+          'impact' => strtolower((string) ($violation['impact'] ?? self::IMPACT_MINOR)),
+          'count' => count($violation['nodes'] ?? []),
+        ];
+      }
+      $pages[] = [
+        'url' => (string) $url,
+        'violations' => $chips,
+        'incomplete' => (int) ($page['incomplete'] ?? 0),
+        'passes' => (int) ($page['passes'] ?? 0),
+        'scenarios' => implode(', ', array_keys($page['scenarios'] ?? [])),
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/AccessibilityTrait.php:1161]: Fixed in b2617a4. `accessibilityAggregatePages()` now keys the per-page scenario set by `feature > scenario`, so two same-titled scenarios from different features no longer collapse into one "Seen in scenarios" entry. This also matches the provenance shown in the per-scenario detail section. Tests updated accordingly.
